inference-engine/tests/functional/plugin/cpu/single_layer_tests/one_hot.cpp

```diff
@@ -3,165 +3,313 @@
+        std::pair<ngraph::helpers::InputLayerType, bool> inputType;
+        InferenceEngine::Precision outPrc;
+        CPUSpecificParams cpuParams;
+        std::tie(inputShape, Axis, inputType, Depth, OnValue, OffValue, outPrc, cpuParams) = this->GetParam();
         std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
-        selectedType = std::string("ref_any_") + inPrc.name();
+        selectedType = std::string("ref_any_I32");
+        outType = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(outPrc);
 
-        auto ngOutPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(outPrc);
-        auto depthConst = ngraph::builder::makeConstant<size_t>(ngraph::element::i32, {}, {depth});
-        auto onConst = ngraph::builder::makeConstant<float>(ngOutPrc, {}, {onValue});
-        auto offConst = ngraph::builder::makeConstant<float>(ngOutPrc, {}, {offValue});
+        init_input_shapes({inputShape});
+        if (inputType.second) {
+            for (auto &target : targetStaticShapes)
+                target.push_back({});
+        }
 
-        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
-        auto inputParams = ngraph::builder::makeParams(ngPrc, { inputShape });
-
-        auto oneHot = std::make_shared<ngraph::opset5::OneHot>(inputParams.front(), depthConst, onConst, offConst, axis);
-        function = makeNgraphFunction(ngPrc, inputParams, oneHot, "OneHot");
+        function = createFunction(!inputType.second);
+    }
+    void init_ref_function(std::shared_ptr<ov::Function> &funcRef, const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        if (function->get_parameters().size() == 2) {
+            // Generate depth
+            ov::runtime::Tensor tensor = ov::test::utils::create_and_fill_tensor(ov::element::Type_t::i32, {}, 10, 1, 1, time(0));
+            auto *dataPtr = tensor.data<int32_t>();
+            Depth = dataPtr[0];
+            funcRef = createFunction(true);
+        }
+        ngraph::helpers::resize_function(funcRef, targetInputStaticShapes);
+    }
+    void validate() override {
+        if (function->get_parameters().size() == 2) {
+            auto pos = std::find_if(inputs.begin(), inputs.end(),
+                                    [](const std::pair<std::shared_ptr<ov::Node>, ov::runtime::Tensor> &params) {
+                                        return params.first->get_friendly_name() == "ParamDepth";
+                                    });
+            IE_ASSERT(pos != inputs.end());
+            inputs.erase(pos);
+        }
+        SubgraphBaseTest::validate();
+    }
+    std::shared_ptr<ngraph::Function> createFunction(bool depthConst) {
+        auto params = ngraph::builder::makeDynamicParams(ngraph::element::i32, {inputDynamicShapes.front()});
+        params.front()->set_friendly_name("ParamsIndices");
+        auto depth_const = std::make_shared<ngraph::op::Constant>(ngraph::element::i32, ngraph::Shape{ }, Depth);
```

Comment:
maybe better to create one pointer to ov::Node and used it for both param and constant